### ISSUE TITLE
pgroonga: object isn't found: <Sourcesxxxxxxx> when configuring PGroonga with Streaming Replication

### ISSUE THREAD

**Issue by dsavinkov** [pgroonga/pgroonga#34]
### Description

PostgreSQL 9.6.2  installed 
groonga-release-1.2.0-1.noarch.rpm installed
Master-Slave Streaming Replication configured according to this [replication replication doc](https://pgroonga.github.io/reference/replication.html) 

After initial configuration is done, the following 2 observations are found:

1. Observation    
```shell
psql (9.6.2)
Type "help" for help.

devqa=# select id,title from "test".contact where title %% '履歴書';
        id        | title  
------------------+--------
 0030000UJgFfAAL1 | 履歴書
(1 row)

devqa=# update "test".contact set title = '履歴書 - 1' where sfid = '0030000UJgFfAAL1';
UPDATE 1
devqa=# 
devqa=# 
devqa=# select id,title from "test".contact where title %% '履歴書';
ERROR:  operator does not exist: text %% unknown
LINE 1: ...tle from "test".contact where title %% '履歴書...
                                                             ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
devqa=# 
```
This error is gone only when I reconnect to Postgres server again OR after server restart.   

2. Observation.
Even though the error is gone. If I try to run the same query on replica server, I have the following error:

```shell
devqa=# select id,title from "test".contact where title %% '履歴書';
ERROR:  pgroonga: object isn't found: <Sources6034694>
```

My question here what am doing wrong here? and how can it be fixed?

**Comment by kou** [comment]
> My question here what am doing wrong here?

I don't know because you didn't show command lines (and configuration changes) you did and their outputs.

> and how can it be fixed?

Try `RECREATE INDEX contact` on master node.

**Comment by dsavinkov** [comment]
@kou I apologize for the delay. 

> My question here what am doing wrong here?

Here what I did:

With existing PostgreSQL 9.6.2 installation - I did not do `initdb` - and master-slave streaming config, I installed pgroonga plugin on CentOS 6.8 x64:

`wget http://packages.groonga.org/centos/groonga-release-1.2.0-1.noarch.rpm`
`sudo -H yum install -y --nogpgcheck groonga-release-1.2.0-1.noarch.rpm`
`sudo yum install -y --enablerepo=epel groonga` (to install all required libs)
`sudo -H yum install -y postgresql96-pgroonga`
`sudo service postgresql-9.6 restart`
`psql -U <dbuser> -d <database>` & run `CREATE EXTENSION pgroonga;` 


For Streaming Replication I did all standard Postgres steps + all additional ones described in the pgroonga docs and everything seems to be working fine at the beginning.

Observation No.1 which is the error after UPDATE statement is not a big deal as we have 1 DML per transaction in our app, but Observation No. 2 is a real pain as it tends to occur under not very clear circumstances. My sense is when master-replica is configured, and we are trying to add another pgroonga index - this where it breaks the replica.  

`REINDEX INDEX <index name>;` did not help. The only solution I found is to re-initiate replica server again, but `pg_basebackup` is not the option at all.  I will try to do it one more time to have all the steps.
